scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -297,20 +297,26 @@ abstract class LocalScanner(
     }
 
     private fun createMissingArchives(scanResults: Map<Package, List<ScanResult>>, downloadDirectory: File) {
+        val missingArchives = mutableSetOf<Pair<Package, KnownProvenance>>()
+
         scanResults.forEach { (pkg, results) ->
-            val missingArchives = results.mapNotNullTo(mutableSetOf()) { result ->
-                result.provenance.takeUnless { it is KnownProvenance && archiver.hasArchive(it) }
+            results.forEach { (provenance, _, _) ->
+                if (provenance is KnownProvenance && !archiver.hasArchive(provenance)) {
+                    missingArchives += Pair(pkg, provenance)
+                }
             }
+        }
 
-            if (missingArchives.isNotEmpty()) {
-                val pkgDownloadDirectory = downloadDirectory.resolve(pkg.id.toPath())
-                Downloader(downloaderConfig).download(pkg, pkgDownloadDirectory)
+        if (missingArchives.isEmpty()) return
 
-                missingArchives.forEach { provenance ->
-                    if (provenance is KnownProvenance) {
-                        archiveFiles(pkgDownloadDirectory, pkg.id, provenance)
-                    }
-                }
+        missingArchives.forEach { (pkg, provenance) ->
+            val pkgDownloadDirectory = downloadDirectory.resolve(pkg.id.toPath())
+            val downloadProvenance = Downloader(downloaderConfig).download(pkg, pkgDownloadDirectory)
+
+            if (downloadProvenance == provenance) {
```

Comment:
This check includes the download time and will not work as expected until #3924 is merged.

Reply:
Thanks for pointing this out, that was indeed a premature change on my side. Luckily #3924 was just merged.

Reply:
nit: it also compares the `originalVcs` which is also not 100% right, but I'm fixing it with my next PR which will remove the original vcs.

Apart from that it's comparing further attributes from VCS info, like `path` and `revision` which should not be compared (in all cases) either.

Reply:
> nit: it also compares the `originalVcs` which is also not 100% right, but I'm fixing it with my next PR which will remove the original vcs.

Note to myself, that bit just got addressed via https://github.com/oss-review-toolkit/ort/pull/3927.